vyper/functions/functions.py

```diff
@@ -85,6 +90,14 @@ class Floor:
     _inputs = [("value", "decimal")]
     _return_type = "int128"
 
+    def evaluate(self, node):
+        validate_call_args(node, 1)
+        if not isinstance(node.args[0], vy_ast.Decimal):
+            raise InvalidType("Call cannot be folded", node)
```

Comment:
I feel like this should raise "Invalid type for arg" earlier somehow than during folding